Structure_Engine/Query/Description.cs

```diff
@@ -0,0 +1,246 @@
+        }
+
+        /***************************************************/
+
+        public static string Description(this GeneralisedFabricatedBoxProfile profile)
+        {
+            return "GenFabBox " + profile.Height + "x" + profile.Width + "x" + profile.WebThickness + "x" + profile.TopFlangeThickness + "x" + profile.BotFlangeThickness + "x" + profile.TopLeftCorbelWidth + "x" + profile.BotLeftCorbelWidth;
+        }
+
+        /***************************************************/
+
+        public static string Description(this KiteProfile profile)
+        {
+            return "Kite " + Math.Round(profile.Angle1,2) + "x" + profile.Width1 + "x" + profile.Thickness;
+        }
+
+        /***************************************************/
+
+        public static string Description(this FabricatedISectionProfile profile)
+        {
+            return "FabI " + profile.Height + "x" + profile.TopFlangeWidth + "x" + profile.BotFlangeWidth + "x" + profile.WebThickness + "x" + profile.TopFlangeThickness + "x" + profile.BotFlangeThickness;
+        }
+
+        /***************************************************/
+
+        public static string Description(this FreeFormProfile profile)
+        {
+            return "FreeForm";
+        }
+
+        /***************************************************/
+
+        public static string Description(this RectangleProfile profile)
+        {
+            return "Rectangle " + profile.Height + "x" + profile.Width + "x" + profile.CornerRadius;
+        }
+
+        /***************************************************/
+
+        public static string Description(this TSectionProfile profile)
+        {
+            return "T " + profile.Height + "x" + profile.Width + "x" + profile.WebThickness + "x" + profile.FlangeThickness;
+        }
+
+        /***************************************************/
+
+        public static string Description(this GeneralisedTSectionProfile profile)
+        {
+            return "GenT " + profile.Height + "x" + profile.WebThickness + "x" + profile.LeftOutstandWidth + "x" + profile.LeftOutstandThickness + "x" + profile.RightOutstandWidth + "x" + profile.RightOutstandThickness;
+        }
+
+        /***************************************************/
+
+        public static string Description(this TubeProfile profile)
+        {
+            return "Tube " + profile.Diameter + "x" + profile.Thickness;
+        }        
+
+        /***************************************************/
+        /**** Public Methods - Sections                 ****/
+        /***************************************************/
+
+        public static string Description(this SteelSection section)
+        {
+            return "Steel " + section.SectionProfile.IDescription() + " - " + CheckGetMaterialName(section.Material);
```

Comment:
```suggestion
            return "Steel " + section.SectionProfile.IDescription() + CheckGetMaterialName(section.Material);
```
One final change, then good to go!

with the " - " included in the method, we now end up with two "-". So need to either remove for the method, or change to suggestion above everywhere